File: C++/A/939A-lovetriangle.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    int no_planes;
    vector<int> planes;

public:
    void input()
    {
        cin >> no_planes;
        for (int i = 0; i < no_planes; i++)
        {
            int x;
            cin >> x;
            planes.push_back(x);
        }
    }
    bool love_triangle()
    {
        if (planes.empty())
            return false;
        for (int i = 0; i < no_planes; i++)
        {
            if (planes[planes[planes[i] - 1] - 1] == i + 1)
                return true;
        }
        return false;
    }
};

int main()
{
    Solver solution;
    solution.input();
    cout << ((solution.love_triangle()) ? "YES\n" : "NO\n");
}

File: Python/939A-lovetriangle.py
class Solver:
    def __init__(self):
        self.no_planes = 0
        self.planes = []

    def input(self):
        self.no_planes = int(input())
        self.planes = list(map(int, input().split()))

    def love_triangle(self):
        if (len(self.planes) == 0):
            return False
        for i in range(self.no_planes):
            if self.planes[self.planes[self.planes[i]-1]-1] == i + 1:
                return True
        return False


solution = Solver()
solution.input()
print("YES" if solution.love_triangle() else "NO")
